sdk/storage/storage-blob/test/node/highlevel.node.spec.ts

```diff
@@ -732,15 +730,18 @@ describe("Highlevel", () => {
     const aborter = new AbortController();
     const res = await blobClient.download(0, undefined, { abortSignal: aborter.signal });
 
-    let exceptionCaught = false;
-    res.readableStreamBody!.on("error", (err) => {
-      assert.equal(err.name, "AbortError");
-      exceptionCaught = true;
+    const bodyEnded = new Promise<void>((resolve, reject) => {
+      res.readableStreamBody!.on("error", (err) => {
+        if (err.name === "AbortError") {
+          resolve();
+        } else {
+          reject(new Error("Expected readableStreamBody to emit AbortError"));
```

Comment:
I understand it's a rare case. Here may cause an exception thrown out from the test cases, I think we should do our best to avoid this?

Reply:
I'm confused about the concern. Since this promise is `await`ed by the test function, if it rejects with an error Mocha should catch it and fail the test. Or am I misunderstanding your comment?